src/utils/types.ts

```diff
@@ -0,0 +1,7 @@
+export type UnpackedPromise<T> = T extends Promise<infer U> ? U : T
```

Comment:
Don't think we need these anymore?

Reply:
We don't need them, no. The thing is, I will need them whenever I'll do the proxy for the contract. The last step in that type will most definitely be something like:
```
type ContractProxy {....}

export Promisify<ContractProxy>;
```
If it bothers too much, I'll remove it but I left it because it's a quite complicated structure to build and I wouldn't want to spend time on it when I get back to it.